addons/mail/models/mail_message.py

```diff
@@ -940,10 +940,22 @@ def _field_store_repr(self, field_name):
             ]
         return [field_name]
 
-    def _to_store_defaults(self):
+    def _to_store_defaults(self, for_current_user=False):
+        def attachment_extra_fields(message):
+            if for_current_user and message.is_current_user_or_guest_author:
+                return Store.Attr(
+                    "as_author_access_token", value=lambda a: a._get_author_access_token()
+                )
+            return []
         field_names = [
-            # sudo: mail.message - reading attachments on accessible message is allowed
-            Store.Many("attachment_ids", sort="id", sudo=True),
+            Store.Attr(
+                "attachment_ids",
+                lambda m: Store.Many(
```

Comment:
I'm deprecating nesting attr with relations in PR https://github.com/odoo/odoo/pull/212092

We might not want to block this PR, but to keep it mind if we can rebase at some point depending on which PR is merged first.

Then we can simply have "dynamic_fields" (if the name is still the same) here.